### PR TITLE
Add file-sync support to subsidies consumer

### PR DESCRIPTION
**Pull request by aatauil**
## ID
 DGS-84

 ## Description

When syncing subsidy data from loket, we currently do not get the files yet. This change goes and fetches the files from subsidies if needed. 

 ## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Other

 ## How to test
 
 Testing exists of 2 parts, initial sync & subsequent delta syncs. 

**initial sync**
- setup clean instance of app-digitaal-loket
- create some subsidie-formulieren in the app (preferably LEKP 1.0 & fietssubsidy) 
- reset inital sync & create new dump file, see the 3 commands in the wiki
```ex
  match "/job-initiator/*path" do
    forward conn, path, "http://delta-producer-background-jobs-initiator-subsidies/"
  end
```
https://github.com/lblod/app-subsidiedatabank/wiki/How-to-setup#initial-sync-process 
- wait for dump to generate
- setup clean app-subsidiedatabank instance, when synced you should now see the same files in both app-digitaal-loket as the subsidiedatabank:
- be sure to check meta-files as well (table data gets generated into meta files) 

**delta sync**
- if initial sync is ok, go to app-digitaal-loket and create a new subsidy
- after a couple of minutes check if files (bijlages) are also visible in subsidiedatabank
- do this for different file-types & subsidies
- be sure to check meta-files as well (table data gets generated into meta files)

### CODE REVIEW

File: config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  SYNC_BASE_URL, 
+  SYNC_LOGIN_ENDPOINT
+} = require("./config");
+
+
+// Types of operations
+const DOWNLOAD_OPERATION = "download"
+const DELETE_OPERATION = "delete"
+
+// Cookie nescessary to download file from producer
```
Comment (wolfderechter) [inline config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js]:
```suggestion
// Cookie necessary to download file from producer
```

File: config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+/**
+ * Delete a file with specified URI.
+ * @param {string} uri - The URI of the file to be deleted.
+ */
+function deleteFile(uri) {
+  uri = uri.replace(/[<>]/g, "");
+  const fileName = uri.replace('share://', '');
+  const filePath = `/share/${fileName}`;
```
Comment (wolfderechter) [inline config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js]:
```suggestion
  const filePath = uri.replace('share://', '/share/');
```
we don't need filename

**Comment by wolfderechter** [comment]
I added the caching from subsidy-detailview branch: https://github.com/lblod/app-subsidiedatabank/pull/11/commits/dcc5a2718a408bc3b5a353befaf6c08ae189b488.

LGTM

File: config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js (hunk 2 above)
Comment (aatauil) [reply config/subsidies-consumer/subsidy-attachments-dispatching/file-processor.js]: Indeed we dont

**Comment by aatauil** [comment]
> I added the caching from subsidy-detailview branch: [dcc5a27](https://github.com/lblod/app-subsidiedatabank/commit/dcc5a2718a408bc3b5a353befaf6c08ae189b488).
> 
> LGTM

Adding caching is ok, just does not belong in this PR. Best to make a seperate PR for that, reverted the change

**Pull request merged**